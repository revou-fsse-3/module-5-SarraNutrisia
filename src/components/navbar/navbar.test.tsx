import { render } from '@testing-library/react';
import Navbar from '.';
import { useRouter } from 'next/router';

jest.mock('next/router', () => ({
  useRouter: jest.fn(),
}));

describe('Navbar component unit testing', () => {
  test('Navbar renders correctly', () => {
    const mockPush = jest.fn();
    
    (useRouter as jest.Mock).mockImplementation(() => ({
      pathname: '/',
      push: mockPush,
    }));

    const document = render(<Navbar />);
    expect(document).toMatchSnapshot();
  });
});