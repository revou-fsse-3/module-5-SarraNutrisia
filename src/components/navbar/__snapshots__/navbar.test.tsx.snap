// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component unit testing Navbar renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-16sxnlf-MuiPaper-root-MuiAppBar-root"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthXl css-19r6kue-MuiContainer-root"
        >
          <div
            class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-1n7m7vv-MuiSvgIcon-root"
              data-testid="CloudQueueIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19.35 10.04C18.67 6.59 15.64 4 12 4 9.11 4 6.6 5.64 5.35 8.04 2.34 8.36 0 10.91 0 14c0 3.31 2.69 6 6 6h13c2.76 0 5-2.24 5-5 0-2.64-2.05-4.78-4.65-4.96M19 18H6c-2.21 0-4-1.79-4-4s1.79-4 4-4h.71C7.37 7.69 9.48 6 12 6c3.04 0 5.5 2.46 5.5 5.5v.5H19c1.66 0 3 1.34 3 3s-1.34 3-3 3"
              />
            </svg>
            <a
              class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-1vz3bfb-MuiTypography-root"
              href="#app-bar-with-responsive-menu"
            >
              Weather Forecast
            </a>
            <div
              class="MuiBox-root css-1lvtzne"
            >
              <button
                aria-controls="menu-appbar"
                aria-haspopup="true"
                aria-label="account of current user"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeLarge css-1pz7awu-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="MenuIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M3 18h18v-2H3zm0-5h18v-2H3zm0-7v2h18V6z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-o4jk4s-MuiSvgIcon-root"
              data-testid="CloudQueueIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19.35 10.04C18.67 6.59 15.64 4 12 4 9.11 4 6.6 5.64 5.35 8.04 2.34 8.36 0 10.91 0 14c0 3.31 2.69 6 6 6h13c2.76 0 5-2.24 5-5 0-2.64-2.05-4.78-4.65-4.96M19 18H6c-2.21 0-4-1.79-4-4s1.79-4 4-4h.71C7.37 7.69 9.48 6 12 6c3.04 0 5.5 2.46 5.5 5.5v.5H19c1.66 0 3 1.34 3 3s-1.34 3-3 3"
              />
            </svg>
            <a
              class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-1mb2186-MuiTypography-root"
              href="#app-bar-with-responsive-menu"
            >
              Weather Forecast
            </a>
            <div
              class="MuiBox-root css-1t6c9ts"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1q39md6-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                Home
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1q39md6-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                Register
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1q39md6-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                Login
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </header>
    </div>
    <div
      aria-hidden="true"
      class="MuiPopover-root MuiMenu-root MuiModal-root MuiModal-hidden css-10nezku-MuiModal-root-MuiPopover-root-MuiMenu-root"
      id="menu-appbar"
      role="presentation"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root MuiBackdrop-invisible MuiModal-backdrop css-g3hgs1-MuiBackdrop-root-MuiModal-backdrop"
        style="opacity: 0; visibility: hidden;"
      />
      <div
        data-testid="sentinelStart"
        tabindex="-1"
      />
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation8 MuiPopover-paper MuiMenu-paper MuiMenu-paper css-3dzjca-MuiPaper-root-MuiPopover-paper-MuiMenu-paper"
        style="opacity: 0; transform: scale(0.75, 0.5625); visibility: hidden;"
        tabindex="-1"
      >
        <ul
          class="MuiList-root MuiList-padding MuiMenu-list css-6hp17o-MuiList-root-MuiMenu-list"
          role="menu"
          tabindex="-1"
        >
          <li
            class="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-root MuiMenuItem-gutters css-kk1bwy-MuiButtonBase-root-MuiMenuItem-root"
            role="menuitem"
            tabindex="0"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-1uwgr7b-MuiTypography-root"
            >
              Home
            </p>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </li>
          <li
            class="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-root MuiMenuItem-gutters css-kk1bwy-MuiButtonBase-root-MuiMenuItem-root"
            role="menuitem"
            tabindex="-1"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-1uwgr7b-MuiTypography-root"
            >
              Register
            </p>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </li>
          <li
            class="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-root MuiMenuItem-gutters css-kk1bwy-MuiButtonBase-root-MuiMenuItem-root"
            role="menuitem"
            tabindex="-1"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-1uwgr7b-MuiTypography-root"
            >
              Login
            </p>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </li>
        </ul>
      </div>
      <div
        data-testid="sentinelEnd"
        tabindex="-1"
      />
    </div>
  </body>,
  "container": <div>
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-16sxnlf-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthXl css-19r6kue-MuiContainer-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-1n7m7vv-MuiSvgIcon-root"
            data-testid="CloudQueueIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19.35 10.04C18.67 6.59 15.64 4 12 4 9.11 4 6.6 5.64 5.35 8.04 2.34 8.36 0 10.91 0 14c0 3.31 2.69 6 6 6h13c2.76 0 5-2.24 5-5 0-2.64-2.05-4.78-4.65-4.96M19 18H6c-2.21 0-4-1.79-4-4s1.79-4 4-4h.71C7.37 7.69 9.48 6 12 6c3.04 0 5.5 2.46 5.5 5.5v.5H19c1.66 0 3 1.34 3 3s-1.34 3-3 3"
            />
          </svg>
          <a
            class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-1vz3bfb-MuiTypography-root"
            href="#app-bar-with-responsive-menu"
          >
            Weather Forecast
          </a>
          <div
            class="MuiBox-root css-1lvtzne"
          >
            <button
              aria-controls="menu-appbar"
              aria-haspopup="true"
              aria-label="account of current user"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeLarge css-1pz7awu-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="MenuIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 18h18v-2H3zm0-5h18v-2H3zm0-7v2h18V6z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-o4jk4s-MuiSvgIcon-root"
            data-testid="CloudQueueIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19.35 10.04C18.67 6.59 15.64 4 12 4 9.11 4 6.6 5.64 5.35 8.04 2.34 8.36 0 10.91 0 14c0 3.31 2.69 6 6 6h13c2.76 0 5-2.24 5-5 0-2.64-2.05-4.78-4.65-4.96M19 18H6c-2.21 0-4-1.79-4-4s1.79-4 4-4h.71C7.37 7.69 9.48 6 12 6c3.04 0 5.5 2.46 5.5 5.5v.5H19c1.66 0 3 1.34 3 3s-1.34 3-3 3"
            />
          </svg>
          <a
            class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-1mb2186-MuiTypography-root"
            href="#app-bar-with-responsive-menu"
          >
            Weather Forecast
          </a>
          <div
            class="MuiBox-root css-1t6c9ts"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1q39md6-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              Home
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1q39md6-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              Register
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1q39md6-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              Login
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
